scm/github.go

```diff
@@ -575,6 +581,12 @@ func (s *GithubSCM) RejectEnrollment(ctx context.Context, opt *RejectEnrollmentO
 
 // DemoteTeacherToStudent removes user from teachers team, revokes owner status in the organization.
 func (s *GithubSCM) DemoteTeacherToStudent(ctx context.Context, opt *UpdateEnrollmentOptions) error {
+	if !opt.valid() {
+		return ErrMissingFields{
+			Method:  "UpdateEnrollment",
```

Comment:
Wrong method name. That said, I prefer to return just an `fmt.Errorf("missing fields: %+v", opt)` instead of `ErrMissingFields`... Do we use (check for) this error type somewhere? If not, maybe we can just remove it...

Reply:
Added a sub-issue to #864.